### PR TITLE
feat: support MyScale vector database

### LINKED ISSUE #6089: feat: support MyScale vector store
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

nope

### 2. Additional context or comments

I want to add a new vector store type: MyScale, which supports both vector search and semantic search
https://myscale.com/
https://github.com/myscale/myscaledb

MyScale already supports [Langchain](https://python.langchain.com/v0.2/docs/integrations/providers/myscale/) and [LlamaIndex](https://docs.llamaindex.ai/en/stable/examples/vector_stores/MyScaleIndexDemo/)

add Vector Store Configurations:

- MYSCALE_HOST
- MYSCALE_PORT
- MYSCALE_USER
- MYSCALE_PASSWORD
- MYSCALE_DATABASE
- MYSCALE_FTS_PARAMS

### 3. Can you help us with this feature?

- [X] I am interested in contributing to this feature.

### PR DESCRIPTION
**Pull request by jewelzqiu**
# Checklist:

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [x] Please open an issue before creating a PR or link to an existing issue
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

# Description

Support MyScale vector store for knowledge (both vector & semantic search)

- add `clickhouse-connect` 0.7.16

Closes #6089  

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update, included: https://github.com/langgenius/dify-docs/pull/151
- [ ] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# Testing Instructions

- [x] docker compose

change `VECTOR_STORE` environment variable to `myscale` in docker-compose `.env` file

### CODE REVIEW

File: api/core/rag/datasource/vdb/myscale/myscale_vector.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+            ORDER BY {dist} {order} LIMIT {top_k}
+        """
+        try:
+            return [
+                Document(
+                    page_content=r["text"],
+                    metadata=r["metadata"],
+                )
+                for r in self._client.query(sql).named_results()
+            ]
```
Comment (JohnJyong) [inline api/core/rag/datasource/vdb/myscale/myscale_vector.py:139]: the score is not returned, pls follow below example  @jewelzqiu , thanks

```
            score_threshold = kwargs.get("score_threshold", .0) if kwargs.get('score_threshold', .0) else 0.0
            if result.score > score_threshold:
                metadata['score'] = result.score
                doc = Document(
                    page_content=result.payload.get(Field.CONTENT_KEY.value),
                    metadata=metadata,
                )
                docs.append(doc)
```

**Review by bowenliang123 — APPROVED** [review]
LGTM.

File: api/core/rag/datasource/vdb/myscale/myscale_vector.py (hunk 1 above)
Comment (jewelzqiu) [reply api/core/rag/datasource/vdb/myscale/myscale_vector.py:139]: @JohnJyong I have added score threshold filter in the latest commit, thx

**Review by JohnJyong — APPROVED** [review]

**Comment by JohnJyong** [comment]
@jewelzqiu thanks for your contribute ， thanks

**Pull request merged**